### PR TITLE
feat: [PL-27279]: added DTOs, entities and mappers for azure blob connector

### PR DESCRIPTION
**Pull request by tiwaritejas**
## Describe your changes
Added DTOs, entities and mappers for Azure Blob Connector

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38617)

### CODE REVIEW

File: 440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/azureblobconnector/AzureBlobConnector.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@Entity(value = "connectors", noClassnameStored = true)
+@Persistent
+@TypeAlias("io.harness.connector.entities.embedded.azureblobconnector.AzureBlobConnector")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AzureBlobConnector extends Connector {
+  String clientId;
+  String tenantId;
+  String secretKeyRef;
+  String containerURL;
+  boolean isDefault;
```
Comment (meenakshi-harness) [inline 440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/azureblobconnector/AzureBlobConnector.java:45]: Non of these fields are required?

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/azureblobconnector/AzureBlobConnectorDTO.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(name = "AzureBlobConnector", description = "Returns configuration details for the Azure Blob Secret Manager.")
+public class AzureBlobConnectorDTO extends ConnectorConfigDTO implements DelegateSelectable {
+  @Schema(description = SecretManagerDescriptionConstants.AZURE_CLIENT_ID) @NotNull private String clientId;
+  @SecretReference
+  @ApiModelProperty(dataType = "string")
+  @NotNull
+  @Schema(description = SecretManagerDescriptionConstants.AZURE_SECRET_KEY)
+  private SecretRefData secretKey;
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/azureblobconnector/AzureBlobConnectorDTO.java:51]: Should be `secretKeyRef`  same as in Entity class.

File: 970-ng-commons/src/main/java/io/harness/SecretManagerDescriptionConstants.java [hunk 3]
```diff
@@ -63,4 +63,12 @@ public class SecretManagerDescriptionConstants {
       "This is the authentication token used to connect underlying secret manager.";
   public static final String GOOGLE_SECRET_MANAGER_CREDENTIALS =
       "Reference to the secret containing credentials of IAM service account for Google Secret Manager";
+  public static final String AZURE_CLIENT_ID = "Client/Application ID of the Azure App.";
+  public static final String AZURE_SECRET_KEY =
+      "This is the Harness text secret with the Azure authentication key as its value.";
+  public static final String AZURE_TENANT_ID =
+      "The Azure Active Directory (AAD) directory ID where you created your application.";
```
Comment (meenakshi-harness) [inline 970-ng-commons/src/main/java/io/harness/SecretManagerDescriptionConstants.java:70]: remove `directory` after AAD ?

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/azureblobconnector/AzureBlobConnectorDTO.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  @NotNull @Schema(description = SecretManagerDescriptionConstants.AZURE_TENANT_ID) private String tenantId;
+  @Schema(description = SecretManagerDescriptionConstants.AZURE_STORAGE_CONTAINER_URL)
+  @NotNull
+  private String containerURL;
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/azureblobconnector/AzureBlobConnectorDTO.java:55]: Do we have any Url Validation for container URL?

File: 940-secret-manager-client/src/main/java/io/harness/secretmanagerclient/dto/azureblob/AzureBlobConfigUpdateDTO.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+@NoArgsConstructor
+@ToString(exclude = {"secretKey"})
+@EqualsAndHashCode(callSuper = true)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AzureBlobConfigUpdateDTO extends SecretManagerConfigUpdateDTO {
+  private String clientId;
+  private String secretKey;
+  private String tenantId;
+  private String containerURL;
+  private boolean isDefault;
```
Comment (meenakshi-harness) [inline 940-secret-manager-client/src/main/java/io/harness/secretmanagerclient/dto/azureblob/AzureBlobConfigUpdateDTO.java:39]: Add notnull validations where needed

File: 940-secret-manager-client/src/main/java/io/harness/secretmanagerclient/dto/azureblob/AzureBlobConfigUpdateDTO.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+@Getter
+@Setter
+@SuperBuilder
+@NoArgsConstructor
+@ToString(exclude = {"secretKey"})
+@EqualsAndHashCode(callSuper = true)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AzureBlobConfigUpdateDTO extends SecretManagerConfigUpdateDTO {
+  private String clientId;
+  private String secretKey;
```
Comment (meenakshi-harness) [inline 940-secret-manager-client/src/main/java/io/harness/secretmanagerclient/dto/azureblob/AzureBlobConfigUpdateDTO.java:36]: secretKeyRef ?

File: 940-secret-manager-client/src/main/java/io/harness/secretmanagerclient/dto/azureblob/AzureBlobConfigDTO.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+@Getter
+@Setter
+@SuperBuilder
+@NoArgsConstructor
+@ToString(exclude = {"secretKey"})
+@EqualsAndHashCode(callSuper = true)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AzureBlobConfigDTO extends SecretManagerConfigDTO {
+  private String clientId;
+  private String secretKey;
```
Comment (meenakshi-harness) [inline 940-secret-manager-client/src/main/java/io/harness/secretmanagerclient/dto/azureblob/AzureBlobConfigDTO.java:37]: same

File: 930-delegate-tasks/src/main/java/software/wings/beans/AzureBlobConfig.java [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+  private static final String TASK_SELECTORS = "Task Selectors";
+  public static final String AZURE_BLOB_VALIDATION_URL = "harnessAzureBlobValidation";
+  @Attributes(title = "Name", required = true) @NotEmpty private String name;
+
+  @Attributes(title = "Azure Client Id", required = true) @NotEmpty private String clientId;
+
+  @Attributes(title = "Azure Secret Id", required = true)
+  @NotEmpty
+  @Encrypted(fieldName = "azure_secret_id")
+  private String secretKey;
```
Comment (meenakshi-harness) [inline 930-delegate-tasks/src/main/java/software/wings/beans/AzureBlobConfig.java:74]: same

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 12 of 12 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38617)* status: 13 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[`440-connector-nextgen/src/main/java/io/harness/connector/mappers/secretmanagermapper/AzureBlobDTOToEntity.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/38617#-NEeydb1-zO8XcS7OC5A:-NEeydb1-zO8XcS7OC5B:bqmigv7) ([raw file](https://github.com/harness/harness-core/blob/8f37eb25b5333decfd617cf786c479fa51efbf45/440-connector-nextgen/src/main/java/io/harness/connector/mappers/secretmanagermapper/AzureBlobDTOToEntity.java#L21)):*
> ```Java
> public class AzureBlobDTOToEntity implements ConnectorDTOToEntityMapper<AzureBlobConnectorDTO, AzureBlobConnector> {
>   @Override
>   public AzureBlobConnector toConnectorEntity(AzureBlobConnectorDTO connectorDTO) {
> ```

Add UT
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/mappers/secretmanagermapper/AzureBlobEntityToDTO.java` line 17 at r2](https://reviewable.io/reviews/harness/harness-core/38617#-NEf-7hH-wUcpb9jFiQ2:-NEf-7hH-wUcpb9jFiQ3:bqmigv7) ([raw file](https://github.com/harness/harness-core/blob/8f37eb25b5333decfd617cf786c479fa51efbf45/440-connector-nextgen/src/main/java/io/harness/connector/mappers/secretmanagermapper/AzureBlobEntityToDTO.java#L17)):*
> ```Java
> public class AzureBlobEntityToDTO implements ConnectorEntityToDTOMapper<AzureBlobConnectorDTO, AzureBlobConnector> {
>   @Override
>   public AzureBlobConnectorDTO createConnectorDTO(AzureBlobConnector connector) {
> ```

Add UT
___
*[`930-delegate-tasks/src/main/java/io/harness/mappers/AzureBlobConfigMapper.java` line 37 at r2](https://reviewable.io/reviews/harness/harness-core/38617#-NEf-OAm2ZPdXh6xIL2b:-NEf-OAm2ZPdXh6xIL2c:b-ctxcfs) ([raw file](https://github.com/harness/harness-core/blob/8f37eb25b5333decfd617cf786c479fa51efbf45/930-delegate-tasks/src/main/java/io/harness/mappers/AzureBlobConfigMapper.java#L37)):*
> ```Java
>                                           .delegateSelectors(azureBlobConfigDTO.getDelegateSelectors())
>                                           .build();
>     azureBlobConfig.setNgMetadata(ngMetaDataFromDto(azureBlobConfigDTO));
> ```

these can set using builder itself
___
*[`954-connector-beans/src/main/java/io/harness/delegate/beans/connector/azureblobconnector/AzureBlobConnectorDTO.java` line 46 at r2](https://reviewable.io/reviews/harness/harness-core/38617#-NEf2i618Sij5xHmjbjM:-NEf2i618Sij5xHmjbjN:b-3fy7zz) ([raw file](https://github.com/harness/harness-core/blob/8f37eb25b5333decfd617cf786c479fa51efbf45/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/azureblobconnector/AzureBlobConnectorDTO.java#L46)):*
> ```Java
> @Schema(name = "AzureBlobConnector", description = "Returns configuration details for the Azure Blob Secret Manager.")
> public class AzureBlobConnectorDTO extends ConnectorConfigDTO implements DelegateSelectable {
>   @Schema(description = SecretManagerDescriptionConstants.AZURE_CLIENT_ID) @NotNull private String clientId;
> ```

should validate for empty value as well

**Pull request closed without merging**